src/UnderlineNav/UnderlineNav.tsx

```diff
@@ -0,0 +1,173 @@
+import React, {useRef, forwardRef, useCallback, useState, useLayoutEffect, MutableRefObject} from 'react'
+import Box from '../Box'
+import {merge, SxProp, BetterSystemStyleObject} from '../sx'
+import {UnderlineNavContext} from './UnderlineNavContext'
+import {ActionMenu} from '../ActionMenu'
+import {ActionList} from '../ActionList'
+import {useResizeObserver} from '../hooks/useResizeObserver'
+
+/*
+Todo:
+1. Convert context to have array of all items = done
+2. Convert UnderlineNav.Link to ActionList.Item - done
+3. Calculate combined width of items to be 50% of total width available - done
+4. fix the dropdown button - done
+5. figure out horizontal scroll - done
+6. Overflow on resize - done
+7. pass typescript - done
```

Comment:
Non-blocking nitpick: Could we remove these comments since they're done?

Reply:
Sure! Had it there for personal accountability. Will be deleting these lines.